activerecord/lib/active_record/integration.rb

```diff
@@ -96,7 +96,11 @@ def cache_key(*timestamp_names)
     # Note, this method will return nil if ActiveRecord::Base.cache_versioning is set to
     # +false+ (which it is by default until Rails 6.0).
     def cache_version
-      if cache_versioning && timestamp = try(:updated_at)
+      return unless cache_versioning
+
+      if timestamp = try(:updated_at_before_type_cast)
+        timestamp
```

Comment:
Will this change the format of the cache key and thus invalidate all the cache entries?

Reply:
I don't think we've exposed a public API for checking individual attributes, but `@attributes["updated_at"].has_been_read?` would do it.

Reply:
I guess having different behavior based on the status of the object would lead to cache bloat as sometimes the key would be different values. So that's not a great idea. I'm mostly trying to brainstorm on ways to minimize the cache invalidation issue. Having talked this one through with you i don't think it's a good path forward.

Reply:
I think to go forward with this change we need to accept that it will invalidate existing caches.